packages/cli/index.js

```diff
@@ -129,7 +129,7 @@ program
     const addHandler = (filePath) => {
       logger.info(`[${new Date().toLocaleTimeString()}] Reload for file add: ${filePath}`);
       Promise.resolve('').then(() => {
-        if (fsUtil.isSourceFile(filePath) || site.isPluginSourceFile(filePath)) {
+        if (site.isDependencyOfPage(filePath)) {
```

Comment:
In reference to https://github.com/MarkBind/markbind/pull/1337#issuecomment-683319939
@KendrickAng, perhaps the confusion was here

Currently we are determining whether to run the page rebuilding process based on whether the file change has a "source file extension". but there is no such thing, because even files with source file extensions can be **assets** (e.g. a raw `.html` file you don't want processed)

This PR refactors page dependency checking into reusable methods and uses these everywhere instead, which is hopefully clearer!

Reply:
Hmm.. so just to clarify:
1. A source file is one that is parsed by Markbind, and also marked for live reload (in the old version of the code).
2. An asset is one that isn't parsed by Markbind

What exactly is defined a `dependency` in markbind? Is it the file `_variables.md` or files inserted using `<include>`?
Or can I simply define it as all the other files that a particular file we are parsing in markbind need, in order to be parsed fully?

Reply:
> A source file is one that is parsed by Markbind, and also marked for live reload (in the old version of the code).
> An asset is one that isn't parsed by Markbind

yes, rather processed, we mostly rely on self-patched external packages for parsing

> What exactly is defined a dependency in markbind?
> Or can I simply define it as all the other files that a particular file we are parsing in markbind need, in order to be parsed fully?

anything a page _references_, recursively, which includes both examples you mentioned